map_gen/maps/danger_ores/modules/main_ores_circles.lua

```diff
@@ -0,0 +1,68 @@
+local Helper = require 'map_gen.maps.danger_ores.modules.helper'
+local b = require 'map_gen.shared.builders'
+local table = require 'utils.table'
+
+local binary_search = table.binary_search
+local bnot = bit32.bnot
+local sqrt = math.sqrt
+local floor = math.floor
+local value = b.manhattan_value
+
+return function(config)
+    local main_ores = config.main_ores
+    local main_ores_split_count = config.main_ores_split_count or 1
+    local start_ore_shape = config.start_ore_shape
+
+    main_ores = Helper.split_ore(main_ores, main_ores_split_count)
+
+    if config.main_ores_shuffle_order then
+      table.shuffle_table(main_ores)
+    end
+
+    local function condition_factory(ore_name)
+      local scale = config.circle_scale or 1
+      --local randomize_scale = config.randomize_scale or false
+      local ore_table = {}
+      for _, ore_data in pairs(main_ores) do
+        table.insert(ore_table, {name = ore_data.name, weight = ore_data.weight})
+      end
+      local weighted = b.prepare_weighted_array(ore_table)
+      return function(x, y, _)
+        local i =  floor(sqrt(x * x + y * y) * scale) % weighted.total + 1
```

Comment:
```suggestion
        local i =  floor(sqrt(x * x + y * y) * scale + 1) % weighted.total + 1
```
So the first ring is iron rather than coal. It's minor but makes the early game slightly easier.

Reply:
Ignore this, I'm dumb. This won't work when shuffle ore is being used.

Reply:
Also at the moment it just works out to be coal since technically I am building from the center of the map. If you change the throttle or types, weights and/or order in the supplied ``ore_table`` it will just change. It is also the reason that first ring usually does not have same width as the other rings as it gets clipped by the spawn and the starting ores.